cluster_experiments/experiment_analysis.py

```diff
@@ -111,6 +114,25 @@ def get_point_estimate(self, df: pd.DataFrame) -> float:
         self._data_checks(df=df)
         return self.analysis_point_estimate(df)
 
+    def pvalue_based_on_hypothesis(self, model_result) -> float:
+        """Returns the p-value of the analysis
+        Arguments:
+            model_result: statsmodels result object
+            verbose (Optional): bool, prints the regression summary if True
+
+        """
+        treatment_effect = model_result.params[self.treatment_col]
+        p_value_half = model_result.pvalues[self.treatment_col] / 2
+
+        if self.hypothesis == "less":
+            p_value = p_value_half if treatment_effect <= 0 else 1 - p_value_half
+        elif self.hypothesis == "greater":
+            p_value = p_value_half if treatment_effect >= 0 else 1 - p_value_half
+        elif self.hypothesis == "two-sided":
+            p_value = model_result.pvalues[self.treatment_col]
```

Comment:
if we do this, I understand we are not using enum, right? Then I would remove the Enum code

Reply:
also, an else clause is missing here raising an error